From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tobias Fischer <tobias.fischer@qut.edu.au>
Date: Fri, 30 Oct 2026 18:21:36 +1000
Subject: [PATCH 2/4] Add OPTION_USE_SYSTEM_SHAPE_LIB to use an installed
 ShapeLib

find_package(shapelib) instead of the bundled extern/shapelib; off by default.
---
 libs/qCC_io/CMakeLists.txt | 20 +++++++++++++++++---
 1 file changed, 17 insertions(+), 3 deletions(-)

diff --git a/libs/qCC_io/CMakeLists.txt b/libs/qCC_io/CMakeLists.txt
index afd11b56..b1a4677b 100644
--- a/libs/qCC_io/CMakeLists.txt
+++ b/libs/qCC_io/CMakeLists.txt
@@ -16,10 +16,24 @@ if( ${OPTION_USE_DXF_LIB} )
 endif()
 
 option( OPTION_USE_SHAPE_LIB "Build with ShapeLib (SHP files support)" ON )
+option( OPTION_USE_SYSTEM_SHAPE_LIB "Use an installed ShapeLib instead of the bundled copy" OFF )
 if( ${OPTION_USE_SHAPE_LIB} )
-	set( SHAPELIB_LIB_DIR "extern/shapelib" CACHE PATH "shapelib lib path" )
-	add_subdirectory( "${SHAPELIB_LIB_DIR}" )
-	target_link_libraries( ${PROJECT_NAME} PUBLIC shapelib )
+	if( ${OPTION_USE_SYSTEM_SHAPE_LIB} )
+		find_package( shapelib CONFIG QUIET )
+		if ( TARGET shapelib::shp )
+			target_link_libraries( ${PROJECT_NAME} PUBLIC shapelib::shp )
+		else()
+			# no (usable) CMake package: the header and library will do
+			find_path( SHAPELIB_INCLUDE_DIR shapefil.h REQUIRED )
+			find_library( SHAPELIB_LIBRARY NAMES shp shapelib REQUIRED )
+			target_include_directories( ${PROJECT_NAME} PUBLIC ${SHAPELIB_INCLUDE_DIR} )
+			target_link_libraries( ${PROJECT_NAME} PUBLIC ${SHAPELIB_LIBRARY} )
+		endif()
+	else()
+		set( SHAPELIB_LIB_DIR "extern/shapelib" CACHE PATH "shapelib lib path" )
+		add_subdirectory( "${SHAPELIB_LIB_DIR}" )
+		target_link_libraries( ${PROJECT_NAME} PUBLIC shapelib )
+	endif()
 	target_compile_definitions( ${PROJECT_NAME} PUBLIC CC_SHP_SUPPORT )
 endif()
 
-- 
2.53.0

